src/Components/Web/src/Forms/InputLargeTextArea/InputLargeTextArea.cs

```diff
@@ -0,0 +1,149 @@
+        /// <param name="cancellationToken">The <see cref="System.Threading.CancellationToken"/> used to relay cancellation of the request.</param>
+        /// <returns>A <see cref="System.IO.TextReader"/> which facilitates reading of the textarea value.</returns>
+        public async ValueTask<StreamReader> GetTextAsync(int maxLength = 32_000, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var streamRef = await JSRuntime.InvokeAsync<IJSStreamReference>(InputLargeTextAreaInterop.GetText, cancellationToken, _inputLargeTextAreaElement);
+                var stream = await streamRef.OpenReadStreamAsync(maxLength, cancellationToken);
+                var streamReader = new StreamReader(stream);
+                return streamReader;
+            }
+            catch (JSException jsException)
+            {
+                // Special casing support for empty textareas. Due to security considerations
+                // 0 length streams/textareas aren't permitted from JS->.NET Streaming Interop.
+                if (jsException.InnerException is ArgumentOutOfRangeException)
+                {
+                    return StreamReader.Null;
+                }
+
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Sets the textarea value asyncronously.
+        /// </summary>
+        /// <param name="streamWriter">A <see cref="System.IO.StreamWriter"/> used to set the value of the textarea.</param>
+        /// <param name="leaveTextAreaEnabled">Don't disable the textarea while settings the new textarea value from the stream.</param>
+        /// <param name="cancellationToken">The <see cref="System.Threading.CancellationToken"/> used to relay cancellation of the request.</param>
+        public async ValueTask SetTextAsync(StreamWriter streamWriter, bool leaveTextAreaEnabled = false, CancellationToken cancellationToken = default)
+        {
+            if (streamWriter.Encoding is not UTF8Encoding)
+            {
+                throw new FormatException($"Expected {typeof(UTF8Encoding)}, got ${streamWriter.Encoding}");
+            }
+
+            try
+            {
+                if (!leaveTextAreaEnabled)
+                {
+                    await JSRuntime.InvokeVoidAsync(InputLargeTextAreaInterop.EnableTextArea, cancellationToken, _inputLargeTextAreaElement, /* disabled: */ true);
+                }
+
+                // Ensure we're reading from the beginning of the stream,
+                // the StreamWriter.BaseStream.Position will be at the end by default
+                var stream = streamWriter.BaseStream;
+                if (stream.Position != 0)
+                {
+                    if (!stream.CanSeek)
+                    {
+                        throw new NotSupportedException("Unable to read from the beginning of the stream.");
```

Comment:
Should we make this check before disabling the input?

Reply:
Also `NotSupportedException` is a fairly odd exception to throw here. `InvalidOperationException` would be preferrable. That said, what happens if we indiscriminately seek to the beginning all the time? It makes the error experience more consistent and Stream would throw a reasonable error if it's not seekable.

```C#
try
{
   streamWriter.BaseStream.Position = 0;
  if (!leaveTextAreaEnabled)
  {
      await JSRuntime.InvokeVoidAsync(InputLargeTextAreaInterop.EnableTextArea, cancellationToken, _inputLargeTextAreaElement, /* disabled: */ true);
  }
}
```